File: ITsupportApp_BACKEND/src/main/java/com/itsupport/itsupport_backend/model/Dto/TicketSupportDTO.java
package com.itsupport.itsupport_backend.model.Dto;

import com.itsupport.itsupport_backend.model.Entity.Equipement;
import com.itsupport.itsupport_backend.model.Entity.Panne;
import com.itsupport.itsupport_backend.model.Entity.Technicien;
import com.itsupport.itsupport_backend.model.Entity.Utilisateur;
import com.itsupport.itsupport_backend.model.Enum.EtatTicket;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class TicketSupportDTO {

    private Long id;
    private String description;
    private LocalDateTime dateCreation;
    private EtatTicket etat;
    private Technicien technicien;
    private Utilisateur utilisateur;
    private Panne panne;
    private Equipement equipement;
}

File: ITsupportApp_FRONTEND/src/app/services/models/ticket-support-dto.ts
/* tslint:disable */
/* eslint-disable */
import { Equipement } from '../models/equipement';
import { Panne } from '../models/panne';
import { Technicien } from '../models/technicien';
import { Utilisateur } from '../models/utilisateur';
export interface TicketSupportDto {
  dateCreation?: string;
  description: string;
  equipement: Equipement;
  etat: 'A_FAIRE' | 'EN_COURS' | 'TERMINE';
  id?: number;
  panne: Panne;
  technicien: Technicien ;
  utilisateur: Utilisateur;
}
